src/docs/sdk/performance/opentelemetry/span-protocol.mdx

```diff
@@ -0,0 +1,342 @@
+var canonicalCodesHTTPMap = map[string]sentry.SpanStatus{
+	"400": sentry.SpanStatusFailedPrecondition, // SpanStatusInvalidArgument, SpanStatusOutOfRange
+	"401": sentry.SpanStatusUnauthenticated,
+	"403": sentry.SpanStatusPermissionDenied,
+	"404": sentry.SpanStatusNotFound,
+	"409": sentry.SpanStatusAborted, // SpanStatusAlreadyExists
+	"429": sentry.SpanStatusResourceExhausted,
+	"499": sentry.SpanStatusCanceled,
+	"500": sentry.SpanStatusInternalError, // SpanStatusDataLoss, SpanStatusUnknown
+	"501": sentry.SpanStatusUnimplemented,
+	"503": sentry.SpanStatusUnavailable,
+	"504": sentry.SpanStatusDeadlineExceeded,
+}
+
+// canonicalCodesGrpcMap maps some GRPC codes to Sentry's span statuses. See description in grpc documentation.
+var canonicalCodesGrpcMap = map[string]sentry.SpanStatus{
+	"1":  sentry.SpanStatusCanceled,
+	"2":  sentry.SpanStatusUnknown,
+	"3":  sentry.SpanStatusInvalidArgument,
+	"4":  sentry.SpanStatusDeadlineExceeded,
+	"5":  sentry.SpanStatusNotFound,
+	"6":  sentry.SpanStatusAlreadyExists,
+	"7":  sentry.SpanStatusPermissionDenied,
+	"8":  sentry.SpanStatusResourceExhausted,
+	"9":  sentry.SpanStatusFailedPrecondition,
+	"10": sentry.SpanStatusAborted,
+	"11": sentry.SpanStatusOutOfRange,
+	"12": sentry.SpanStatusUnimplemented,
+	"13": sentry.SpanStatusInternalError,
+	"14": sentry.SpanStatusUnavailable,
+	"15": sentry.SpanStatusDataLoss,
+	"16": sentry.SpanStatusUnauthenticated,
+}
+
+code := spanStatus.Code()
+if code < 0 || int(code) > 2 {
+    return sentry.SpanStatusUnknown, fmt.Sprintf("error code %d", code)
+}
+httpCode, foundHTTPCode := tags["http.status_code"]
+grpcCode, foundGrpcCode := tags["rpc.grpc.status_code"]
+var sentryStatus sentry.SpanStatus
+switch {
+case code == 1 || code == 0:
+    sentryStatus = sentry.SpanStatusOK
+case foundHTTPCode:
+    httpStatus, foundHTTPStatus := canonicalCodesHTTPMap[httpCode]
+    switch {
+    case foundHTTPStatus:
+        sentryStatus = httpStatus
+    default:
+        sentryStatus = sentry.SpanStatusUnknown
+    }
+case foundGrpcCode:
+    grpcStatus, foundGrpcStatus := canonicalCodesGrpcMap[grpcCode]
+    switch {
+    case foundGrpcStatus:
+        sentryStatus = grpcStatus
+    default:
+        sentryStatus = sentry.SpanStatusUnknown
+    }
+default:
+    sentryStatus = sentry.SpanStatusUnknown
+}
+return sentryStatus
+```
+
+### Span Events
+
+OpenTelemetry, has the concept of [Span Events](https://github.com/open-telemetry/opentelemetry-proto/blob/724e427879e3d2bae2edc0218fff06e37b9eb46e/opentelemetry/proto/trace/v1/trace.proto#L193-L214). As per the spec:
+
+> An event is a human-readable message on a span that represents “something happening” during it’s lifetime
+
+In Sentry, we have two options for how to treat span events. First, we can add them as breadcrumbs to the transaction the span belongs to. Second, we can create an artificial "point-in-time" span (a span with 0 duration), and add it to the span tree. TODO on what approach we take here.
```

Comment:
Adding a _point-in-time_ span this way seems a bit hacky to me.

When you say "_we can add them as breadcrumbs to the transaction_", you mean we can add them as child transactions, not via `addBreadcrumb()`

Maybe clarifying this a bit so people (like me) don't mistake spans for breadcrumbs :)

Reply:
I suggest to add them as both, breadcrumbs and point-in-time-spans.
The Ruby docs of oTel span events [1] state that `characteristic of events is that their timestamps are displayed as offsets from the beginning of the span`. This suggested we display them in our performance waterfall diagram. 
And adding them to the breadcrumbs makes sense, because this is a fundamental Sentry feature and existing users are used to it. 

1: https://opentelemetry.io/docs/instrumentation/ruby/events/

Reply:
Let's table this discussion - we can leave this as a TODO item here and evaluate this once we only really need to (like when the first Ruby SDK part is ready to review).